sdk/core/allocator/main.cc

```diff
@@ -387,18 +387,25 @@ namespace
 	}
 
 	/**
-	 * Unseal an allocator capability and return it.  Returns `nullptr` if this
-	 * is not a heap capability.
+	 * Unseal an allocator capability and return it.
+	 *
+	 * Returns a pair, with `nullptr` as the first element if this is not a
+	 * heap capability.
+	 *
+	 * The permission bits from the sealed capability are extracted and returned
+	 * in the second element of the pair.
 	 */
-	PrivateAllocatorCapabilityState *
+	std::pair<PrivateAllocatorCapabilityState *, uint8_t>
```

Comment:
Minor nit, but making this a `uint8_t` will generate some explicit masking instructions, whereas making it an `int` will likely result in very slightly smaller code.